documentation/metrics.md

```diff
@@ -0,0 +1,78 @@
+
+In addition to metrics that downstream services may choose to make available via the status service, the status
+service itself ships with some basic JVM metrics that can be useful for monitoring the process as a whole.
+
+Here is an example of what the status service's own status callback returns at `:debug` level:
+
+```json
+    "status-service": {
+        "active_alerts": [],
+        "detail_level": "debug",
+        "service_status_version": 1,
+        "service_version": "0.5.0",
+        "state": "running",
+        "status": {
+            "experimental": {
+                "jvm-metrics": {
+                    "file-descriptors": {
+                        "max": 65536,
+                        "used": 198
+                    },
+                    "gc-stats": {
+                        "PS MarkSweep": {
+                            "count": 5,
+                            "total-time-ms": 657
+                        },
+                        "PS Scavenge": {
+                            "count": 17,
+                            "total-time-ms": 306
+                        }
+                    },
+                    "heap-memory": {
+                        "committed": 1111490560,
+                        "init": 262144000,
+                        "max": 1908932608,
+                        "used": 612812056
+                    },
+                    "non-heap-memory": {
+                        "committed": 265027584,
+                        "init": 2555904,
+                        "max": -1,
+                        "used": 178038080
+                    },
+                    "start-time-ms": 1475685724906,
+                    "up-time-ms": 25466
+                }
+            }
+        }
+    }
+
+```
+
+The most interesting part of the payload above is the data available in the "status" -> "experimental" -> "jvm-metrics"
+map.  Here are some details about the fields available there:
+
+* `heap-memory`: information about the JVM's heap memory usage; this mostly accounts for memory consumed by application code
+** `committed`: the amount of memory that the operating system has allocated to the process
+** `init`: the initial amount of memory that was allocated to the process at startup
+** `max`: the maximum amount of memory that the process will request from the operating system before throwing an OOM error
+** `used`: the amount of memory that is currently being used by the application
+* `non-heap-memory`: same fields as for `heap-memory`, but this refers to native memory used by the JVM itself, along with any memory allocated by native libraries
+* `file-descriptors`:
+** `max`: the maximum number of file descriptors that this process is allowed to open
+** `used`: the current number of file descriptors the process has open
+* `gc-stats`: a map containing key garbage collection statistics for each of the different GC algorithms that are in use by this JVM.  The keys in this map are the GC algorithm names
```

Comment:
Is there any documentation or anything that would be useful to link to as far as what these GC algorithm names are, or is that common enough knowledge?